src/mcts/node.cc

```diff
@@ -95,7 +95,7 @@ class NodeGarbageCollector {
   std::vector<std::unique_ptr<Node>> subtrees_to_gc_ GUARDED_BY(gc_mutex_);
 
   // When true, Worker() should stop and exit.
-  volatile bool stop_ = false;
+  std::atomic<bool> stop_ = {false};
```

Comment:
Maybe instead:
```cpp
std::atomic<bool> stop_{false};
```
I don't have much against copy list initialization, but Scott Meyers doesn't like it for incompatibility with `auto` and recommends against using it. See for example:
<http://scottmeyers.blogspot.com/2015/09/thoughts-on-vagaries-of-c-initialization.html>

Also if `stop_` is atomic, it's more clean to use access functions (`store` and `load`, where you can set memory_order) rather than using it implicitly.

Reply:
What memory order do you want to use then?  I figured default is fine for this and not using store or load explicitly will use the default.